### DISCUSSION TITLE
How to add query to current route without reload component in nuxt 3

### DISCUSSION THREAD

**Discussion by xushnudbek20** [nuxt/nuxt#24062] [Questions] [answered]
### Environment

vue-router nuxt

### Reproduction

https://stackblitz.com/edit/github-1obxwz

### Describe the bug

How to add query to current route without reload component in nuxt 3. 
Below example. If I add some query on current route, It recalling component, mounted working again, In vue hasn't problem like that.

**Comment by TheAlexLichter** [comment, marked as answer]
If you use `<NuxtPage :key="$route.fullPath" />`, the page component will *always re-render when the path changes*.

I'd suggest using `:page-key="route => route.path"` instead.

**Reply by xushnudbek20** [reply]
It's works. Thank you

**Reply by xushnudbek20** [reply]
@manniL If  I insert `:page-key="route => route.path"` instead of `:key="$route.fullPath"`, I got an error. 

 _Failed to execute 'insertBefore' on 'Node': The node before which the new node is to be inserted is not a child of this node._

**Comment by remihuigen** [comment]
@manniL  Using the method mentioned above ``page:loading:start`` is still triggered am I right? It shows the ``<NuxtLoadingIndicator />`` for me...
